### PR TITLE
Refactoring for standard field structure

### PR DESCRIPTION
**Pull request by bmallred**
Issue: #406

### CODE REVIEW

**Comment by floresj** [comment]
This is exciting. :+1: :+1:

File: src/components/Section/Identification/ApplicantBirthPlace/ApplicantBirthPlace.jsx [hunk 1]
```diff
@@ -250,16 +248,15 @@ export default class ApplicantBirthPlace extends ValidationElement {
...
+          </Field>
         </div>
       )
     }
 
     return (
       <div className={klass}>
         {this.options()}
-        <Help id="identification.birthplace.help.city">
+        <Field help="identification.birthplace.help.city">
```
Comment (floresj) [inline src/components/Section/Identification/ApplicantBirthPlace/ApplicantBirthPlace.jsx]: `Field` makes the nerd soul happy.

**Comment by floresj** [comment]
Yesss yessss remove it!

File: src/components/Form/Accordion/Accordion.jsx [hunk 2]
```diff
@@ -261,7 +266,7 @@ export default class Accordion extends ValidationElement {
       <div className={klassAppend}>
         {title}
         {message}
-        <div className="eapp-field-wrap">
+        <div>
```
Comment (floresj) [inline src/components/Form/Accordion/Accordion.jsx]: Goodbye field wrap

Comment (bmallred) [inline src/components/Form/Accordion/Accordion.jsx]: So long and thanks for all the :fish:

File: src/components/Form/Reference/Reference.jsx [hunk 3]
```diff
@@ -94,149 +94,138 @@ export default class Reference extends ValidationElement {
...
-        <h3>{i18n.t(`${prefix}reference.heading.address`)}</h3>
-        <p>{i18n.t(`${prefix}reference.para.address`)}</p>
-        <div className="eapp-field-wrap">
-          <Help id={`${prefix}reference.help.address`}>
-            <Address name="Address"
-                     {...this.state.Address}
-                     label={i18n.t(`${prefix}reference.label.address`)}
-                     onUpdate={this.onUpdate.bind(this, 'Address')}
+        <Field title={i18n.t(`${prefix}reference.heading.phone`)}
+               help="${prefix}reference.help.phone"
```
Comment (floresj) [inline src/components/Form/Reference/Reference.jsx]: Should `help` prop be enclosed in \` \` instead of " " ?

Comment (bmallred) [inline src/components/Form/Reference/Reference.jsx]: Good catch... went macro crazy!

File: src/components/Form/Reference/Reference.jsx [hunk 4]
```diff
@@ -94,149 +94,138 @@ export default class Reference extends ValidationElement {
...
+                       onUpdate={this.onUpdate.bind(this, 'LastContact')}
                        onValidate={this.props.onValidate}
                        />
-            <HelpIcon className="phone-help-icon" />
-          </Help>
-        </div>
+        </Field>
+
+        <Field title={i18n.t(`${prefix}reference.heading.relationship`)}
+               help="${prefix}reference.help.relationship"
```
Comment (floresj) [inline src/components/Form/Reference/Reference.jsx]: Swap los "" con \`\`

File: src/components/Section/Family/Relatives/Relative.scss [hunk 5]
```diff
@@ -2,36 +2,16 @@
 
 .relative-item {
   .smaller {
+    display: block;
     font-size: smaller;
+    margin-top: -0.5rem;
   }
 
   .up-a-bit {
```
Comment (floresj) [inline src/components/Section/Family/Relatives/Relative.scss]: I love this class name.

File: src/components/Section/Family/Relatives/Relative.scss [hunk 6]
```diff
@@ -77,18 +58,18 @@
 
   .relative-naturalized label,
   .relative-derived label {
-    width: 30rem;
+    width: 27rem;
 
     > span > div > p {
-      line-height: 1.2;
+      line-height: 1.6;
```
Comment (floresj) [inline src/components/Section/Family/Relatives/Relative.scss]: Does this need a unit like rem? I can't remember for line height. Just double checkin'

Comment (bmallred) [inline src/components/Section/Family/Relatives/Relative.scss]: Nope, according to [W3C](https://www.w3schools.com/cssref/pr_dim_line-height.asp):

> number	A number that will be multiplied with the current font size to set the line height

File: src/components/Section/Financial/Gambling/Gambling.jsx [hunk 7]
```diff
@@ -153,67 +134,53 @@ export default class Gambling extends ValidationElement {
...
+                   titleSize="h4"
+                   help="financial.gambling.help.description">
+              <Textarea name="Description"
+                        className="description"
+                        bind={true}
+                        />
+            </Field>
+
+            <Field title={i18n.t('financial.gambling.heading.actions')}
+                   titleSize="h4"
```
Comment (floresj) [inline src/components/Section/Financial/Gambling/Gambling.jsx]: Sweet.

File: src/components/Section/History/Employment/Employment.jsx [hunk 8]
```diff
@@ -120,153 +120,125 @@ export class EmploymentItem extends ValidationElement {
...
           />
 
         <Show when={this.showEmployer()}>
-          <div>
-            <h3>{i18n.t(`${prefix}.heading.employer`)}</h3>
-            <div className="eapp-field-wrap">
-              <Help id={`${prefix}.employer.help`}>
-                <Text name="Employment"
+          <Field title={i18n.t(`${prefix}.heading.employer`)}
+                 help="${prefix}.employer.help"
```
Comment (floresj) [inline src/components/Section/History/Employment/Employment.jsx]: Swappy swap "" with some of \`\`

File: src/components/Section/History/Employment/Employment.jsx [hunk 9]
```diff
@@ -120,153 +120,125 @@ export class EmploymentItem extends ValidationElement {
...
+          </Field>
         </Show>
 
         <Show when={this.showEmployed()}>
-          <div>
-            <h3>{i18n.t(`${prefix}.heading.title`)}</h3>
-            <div className="eapp-field-wrap">
-              <Help id={`${prefix}.title.help`}>
-                <Text name="Title"
+          <Field title={i18n.t(`${prefix}.heading.title`)}
```
Comment (floresj) [inline src/components/Section/History/Employment/Employment.jsx]: Stop watching me.

File: src/components/Section/History/Employment/Employment.jsx [hunk 10]
```diff
@@ -120,153 +120,125 @@ export class EmploymentItem extends ValidationElement {
...
         </Show>
 
         <Show when={this.showAssignedDuty()}>
-          <div>
-            <h3>{i18n.t(`${prefix}.heading.dutyStation`)}</h3>
-            <div className="eapp-field-wrap">
-              <Help id={`${prefix}.dutyStation.help`}>
-                <Text name="DutyStation"
+          <Field title={i18n.t(`${prefix}.heading.dutyStation`)}
+                 help="${prefix}.dutyStation.help"
```
Comment (floresj) [inline src/components/Section/History/Employment/Employment.jsx]: Swap "" with \`\`

File: src/components/Section/History/Employment/Employment.jsx [hunk 11]
```diff
@@ -120,153 +120,125 @@ export class EmploymentItem extends ValidationElement {
...
-                  {...this.props.Status}
-                  onUpdate={this.onUpdate.bind(this, 'Status')}
-                  onValidate={this.props.onValidate}
-                />
-                <HelpIcon className="status" />
-              </Help>
-            </div>
-          </div>
+          <Field title={i18n.t(`${prefix}.heading.status`)}
+                 help="${prefix}.status.help"
```
Comment (floresj) [inline src/components/Section/History/Employment/Employment.jsx]: Swap "" with \`\`

File: src/components/Section/History/Employment/Employment.jsx [hunk 12]
```diff
@@ -120,153 +120,125 @@ export class EmploymentItem extends ValidationElement {
...
-                  onUpdate={this.onUpdate.bind(this, 'Address')}
-                  onValidate={this.props.onValidate}
-                  label={i18n.t(`${prefix}.address.label`)}
-                />
-                <HelpIcon className="address"/>
-              </Help>
-            </div>
-          </div>
+          <Field title={i18n.t(`${prefix}.heading.address`)}
+                 help="${prefix}.address.help"
```
Comment (floresj) [inline src/components/Section/History/Employment/Employment.jsx]: Swap "" with \`\`

File: src/components/Section/History/Employment/Employment.jsx (hunk 9 above)
Comment (bmallred) [inline src/components/Section/History/Employment/Employment.jsx]: :eyes:

File: src/components/Section/History/Employment/Employment.jsx [hunk 13]
```diff
@@ -120,153 +120,125 @@ export class EmploymentItem extends ValidationElement {
...
-                  {...this.props.Telephone}
-                  onUpdate={this.onUpdate.bind(this, 'Telephone')}
-                  onValidate={this.props.onValidate}
-                />
-                <HelpIcon className="telephone-icon"/>
-              </Help>
-            </div>
-          </div>
+          <Field title={i18n.t(`${prefix}.heading.telephone`)}
+                 help="${prefix}.telephone.help">
```
Comment (floresj) [inline src/components/Section/History/Employment/Employment.jsx]: I've been told a macro will dominate these.

Comment (bmallred) [inline src/components/Section/History/Employment/Employment.jsx]: qn0f"xi{`<Escape>f"C`}<Escape>0

File: src/components/Section/Legal/Police/Offense.jsx [hunk 14]
```diff
@@ -302,145 +292,139 @@ export default class Offense extends ValidationElement {
...
-              onUpdate={this.updateWasSentenced}
-              onValidate={this.props.onValidate}>
+                    className="offense-sentenced"
+                    value={this.state.WasSentenced}
+                    help="legal.police.help.sentenced"
+                    onUpdate={this.updateWasSentenced}
+                    onValidate={this.props.onValidate}>
             </Branch>
 
             <Show when={this.state.WasSentenced === 'Yes'}>
```
Comment (floresj) [inline src/components/Section/Legal/Police/Offense.jsx]: Almost at the end

**Comment by floresj** [comment]
This is awesome. Only comment is I remember adding some explicit padding for the `ApplicantBirthPlace.scss` for the inputs. With your changes, it looks like that has been normalized and that styling I added a while back probably isn't necessary now.

**Comment by bmallred** [comment]
@floresj yeah, so everything is standardized now for the most part and the spacing on birth place should be consistent.

**Pull request merged**